Improve compatibility with dalton-bundled basis set library

peter-reinholdt opened the pull request:
This PR improves the compatibility with reading the "legacy" basis sets bundled with dalton.
Before, I was able to read around 23/291 of the basis sets bundled with dalton, after these changes around 248/291 can be read.

The PR fixes three problems:
- PHOSPHORUS is misspelled as PHOSPHOROUS in many (~86) of the bundled dalton basis set files
- $ is used as a comment in many legacy dalton basis set files
- For basis sets with many primitives, the contraction coefficients are sometimes split over multiple lines

susilehtola commented on basis_set_exchange/readers/dalton.py:
```diff
@@ -64,7 +65,7 @@ def _parse_electron_lines(basis_lines, bs_data):
             i += 1
 
-    basis_lines = new_basis_lines
+    basis_lines = [line for line in new_basis_lines if line[0] != '$']
```
```suggestion
    basis_lines = helpers.prune_lines(basis_lines, '$')
```

susilehtola commented on basis_set_exchange/readers/dalton.py:
```diff
@@ -95,7 +96,12 @@ def _parse_electron_lines(basis_lines, bs_data):
+            bas_lines = sh_lines[1:]
+            if len(sh_lines[1:]) != nprim:
+                if len(sh_lines[1:]) == 2 * nprim:
+                    bas_lines = [' '.join([sh_lines[1 + i], sh_lines[2 + i]]) for i in range(nprim)]
```
Shouldn't there be an `else` clause here..?

peter-reinholdt replied:
Yes, this was wrong.
Now it should work for also more than a single line split (relevant for some of the Huzinaga basis sets)

susilehtola commented:
Looks like your code doesn't like the Dalton tests included at the moment, it's raising the assertion error. Do you want to fix this yourself?

It would also be good to add in tests for (some of) the files that couldn't be parsed before.

peter-reinholdt commented:
Yes - I think the test for bad files should fail in the right way now.
I have added some test files for the main cases that the PR addresses ("$" comment; P misspelling; split lines)  (not sure if the tests are added correctly)

susilehtola commented:
That's what I assume; the tests appear to loop over all the files in the directory: https://github.com/MolSSI-BSE/basis_set_exchange/blob/0c56fa3b97bd34b750c059f8fef5b18a38c7dcba/basis_set_exchange/tests/test_readers.py#L16 and https://github.com/MolSSI-BSE/basis_set_exchange/blob/0c56fa3b97bd34b750c059f8fef5b18a38c7dcba/basis_set_exchange/tests/test_readers.py#L23

If the tests pass, I'm happy with this. We'll just have to wait on @bennybp to merge

bennybp commented:
Thanks! This looks good to me.

I had noticed the phosphorous vs. phosphorus thing before, but I think I convinced myself that Dalton wasn't actually reading the element names. I guess I was wrong :)

And yes, I do see the new tests being run
